sbat.c

```diff
@@ -120,4 +121,168 @@ EFI_STATUS parse_sbat(char *sbat_base, size_t sbat_size, char *buffer,
 	return efi_status;
 }
 
+EFI_STATUS
+verify_single_entry(struct sbat_entry *entry, struct sbat_var *sbat_var_entry)
+{
+        UINT16 sbat_gen, sbat_var_gen;
+
+        if (strcmp(entry->component_name, sbat_var_entry->component_name) == 0) {
+                dprint(L"component %a has a matching SBAT variable entry, verifying\n",
+                       entry->component_name);
+                /* atoi returns zero for failed conversion, so essentially
+                badly parsed component_generation will be treated as zero
+                */
+                sbat_gen = atoi(entry->component_generation);
+                sbat_var_gen = atoi(sbat_var_entry->component_generation);
+                if ( sbat_gen < sbat_var_gen ) {
+                        dprint(L"component %a, generation %d, was revoked by SBAT variable",
+                                 entry->component_name, sbat_gen);
+                        LogError(L"image did not pass SBAT verification\n");
+                        return EFI_SECURITY_VIOLATION;
+                }
+        }
+        return EFI_SUCCESS;
+}
+
+EFI_STATUS
+verify_sbat(struct sbat *sbat, list_t *sbat_entries)
+{
+	unsigned int i;
+	list_t *pos = NULL;
+	EFI_STATUS efi_status;
+	
+	for (i = 0; i < sbat->size; i++) {
+		struct sbat_entry *entry = entry = sbat->entries[i];
```

Comment:
why `struct sbat_entry *entry = entry = sbat->entries[i];` and not just `struct sbat_entry *entry =  sbat->entries[i];` ? 

Probably makes more sense to define the variable outside of the for loop though.